satpy/tests/test_demo.py

```diff
@@ -192,3 +192,99 @@ def test_ahi_partial_download(self):
         from tempfile import gettempdir
         files = download_typhoon_surigae_ahi(base_dir=gettempdir(), segments=[4, 9], channels=[1, 2, 3])
         assert len(files) == 6
+
+
+class _FakeRequest:
+    """Fake object to act like a requests return value when downloading a zip file."""
+
+    def __init__(self, url, stream=None):
+        self._filename = os.path.basename(url)
+        self.headers = {}
+        del stream  # just mimicking requests 'get'
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        return
+
+    def raise_for_status(self):
+        return
+
+    def _get_fake_viirs_sdr_bytesio(self):
+        filelike_obj = io.BytesIO()
+        filelike_obj.write(self._filename.encode("ascii"))
+        filelike_obj.seek(0)
+        return filelike_obj
+
+    def iter_content(self, chunk_size):
+        """Return generator of 'chunk_size' at a time."""
+        bytes_io = self._get_fake_viirs_sdr_bytesio()
+        x = bytes_io.read(chunk_size)
+        while x:
+            yield x
+            x = bytes_io.read(chunk_size)
+
+
+class TestVIIRSSDRDemoDownload:
+    """Test VIIRS SDR downloading."""
+
+    ALL_BANDS = ("I01", "I02", "I03", "I04", "I05",
+                 "M01", "M02", "M03", "M04", "M05", "M06", "M07", "M08", "M09", "M10",
+                 "M11", "M12", "M13", "M14", "M15", "M16",
+                 "SVDNB",
```

Comment:
In the download function default parameter, it read "DNB", not "SVDNB" ?
```suggestion
                 "DNB",
```

Reply:
No, this fails if it is DNB because it matches `SVDNB` and `GDNBO`. I could rename the variable to make it more clear. Something like `FILENAME_PREFIX` and then include `SV` in all of these.